src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt

```diff
@@ -18,40 +19,38 @@ class GovNotifyEmailService(
-      is OffenderReleasedCourtEmail -> send(email, emailTemplates.offenderReleasedCourt)
-      is OffenderReleasedPrisonEmail -> send(email, emailTemplates.offenderReleasedPrison)
-      is OffenderReleasedPrisonNoCourtEmail -> send(email, emailTemplates.offenderReleasedPrisonNoCourt)
-      is OffenderTransferredCourtEmail -> send(email, emailTemplates.offenderTransferredCourt)
-      is OffenderTransferredPrisonEmail -> send(email, emailTemplates.offenderTransferredPrison)
-      is OffenderTransferredPrisonNoCourtEmail -> send(email, emailTemplates.offenderTransferredPrisonNoCourt)
+      is CourtNewBookingEmail -> send(email, emailTemplates.courtNewBooking)
       else -> throw RuntimeException("Unsupported email type ${email.javaClass.simpleName}.")
     }
 
   private fun send(email: Email, templateId: String) =
     runCatching {
-      client.sendEmail(templateId, email.address, email.personalisation, null).notificationId!!
+      client.sendEmail(templateId, email.address, email.personalisation(), null).notificationId!!
     }
       .onSuccess { log.info("EMAIL: sent ${email.javaClass.simpleName} email.") }
       .onFailure { log.info("EMAIL: failed to send ${email.javaClass.simpleName} email.") }
 }
 
-class AppointmentCancelledCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class AppointmentCancelledPrisonEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class AppointmentCancelledPrisonNoCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderReleasedCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderReleasedPrisonEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderReleasedPrisonNoCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderTransferredCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderTransferredPrisonEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
-
-class OffenderTransferredPrisonNoCourtEmail(override val address: String, override val personalisation: Map<String, String?>) : Email()
+class CourtNewBookingEmail(
+  address: String,
+  prisonerFirstName: String,
+  prisonerLastName: String,
+  prisonerNumber: String,
+  date: LocalDate = LocalDate.now(),
+  userName: String,
+  court: String,
+  prison: String,
+  preAppointmentInfo: String = "",
+  mainAppointmentInfo: String,
+  postAppointmentInfo: String = "",
+  comments: String?,
+) : Email(address, prisonerFirstName, prisonerLastName, prisonerNumber, date, comments) {
+  init {
+    addPersonalisation("userName", userName)
```

Comment:
Some of these personalisations may become common across all templates but this should do for starters.